drivers/sensor/stm32/temp_stm32f401x.c

```diff
@@ -135,10 +144,9 @@ static int temp_stm32f401x_sample_fetch(struct device *dev,
 	return 0;
 }
 
-static inline float stm32f401x_temp_c(uint32_t v_sense)
+static inline float stm32f401x_temp_c(float v_sense)
 {
-	float sense = (float)v_sense;
-	return ((sense - STM32F401_V25) / STM32F401_AVG_SLOPE + 25.0f);
+	return ((v_sense - STM32F401_V25) / STM32F401_AVG_SLOPE + 25.0f);
```

Comment:
Sure -- I'm just copy/pasting this from the reference manual. I'll add a pointer to the relevant section.